### PR TITLE
feat(remote read): read labelvalues

### PR DESCRIPTION
**Pull request by caitong93**
@tomwilkie @brian-brazil 
This is for issue https://github.com/prometheus/prometheus/issues/3351
The pr add additional fields in `prompb.ReadRequest` and `prompb.ReadResponse` to allow reading label values from adapters
```
message ReadRequest {
  repeated Query queries = 1;
  string labelName = 2;
}

message ReadResponse {
  // In same order as the request's queries.
  repeated QueryResult results = 1;
  repeated string labelValues = 2;
}

```

### CODE REVIEW

**Comment by tomwilkie** [comment]
Thanks @caitong93! 

I’m not sure we want to add this to the same read endpoint though, it really deserves its own pair of protos for request and response and a fresh URL for it. 

I’d been waiting until I had a chance to move this interface to gRPC to implement it, but if you’re keen could you make the changes to this PR?

**Comment by caitong93** [comment]
@tomwilkie Thanks for review!
I agreed that label values should have its own protos and url. Then I think it will look like

```
message LabelValuesRequest {
  string labelName = 2;
}

message LabelValuesResponse {
  repeated string labelValues = 2;
}
```

the config 

```
remote_read:
 - url: http://remote4/read
   label_values_path: `/labelvalues`
```

**Comment by tomwilkie** [comment]
@caitong93 looking good! Couple of comments:
-  Field names should use underscores as per the style guide (https://developers.google.com/protocol-buffers/docs/style). 
- I would specify a separate URL for the labels values call, not just a extra path - it will be more flexible.
Thanks!

**Comment by caitong93** [comment]
@tomwilkie updated

**Comment by AlekSi** [comment]
Gentle poke for @tomwilkie

**Comment by tomwilkie** [comment]
Thanks @AlekSi.  Code looks good, a bunch of minor nits but nothing major.

Before I address those, on #3351 I said "We'll probably want to move to gRPC before we do this, to make the API handling simpler." - which this PR doesn't do.  I inclined to accept this PR anyway, but I think it would be good to get a second opinion from @juliusv first.

File: config/config.go [hunk 1]
```diff
@@ -703,7 +703,8 @@ type RemoteReadConfig struct {
 
 	// RequiredMatchers is an optional list of equality matchers which have to
 	// be present in a selector to query the remote read endpoint.
-	RequiredMatchers model.LabelSet `yaml:"required_matchers,omitempty"`
+	RequiredMatchers model.LabelSet   `yaml:"required_matchers,omitempty"`
+	LabelValuesURL   *config_util.URL `yaml:"label_values_url,omitempty"`
```
Comment (brian-brazil) [inline config/config.go]: If you're adding a new field please update the documentation.

**Comment by caitong93** [comment]
@tomwilkie 
I didn't understand `move to gRPC` at first. But now I know it is `moving remote read and write from http to gRPC`. I would like to give a try.  I can make changes to this pr, or sending another pr after this one. WDYT ?

**Pull request closed without merging**